modules/javafx.graphics/src/main/native-prism-d3d/hlsl/psConstants.h

```diff
@@ -25,12 +25,13 @@
 
 // see D3DPhongShader.h
 
-static const int numMaxLights = 5;
+static const int numMaxLights = 3;
```

Comment:
I know we only support 3 lights, but I note that the vertex shader structures in `vsConstants.h` also have room for 5, and that number was unchanged by this PR. Have you ensured that the mismatch won't cause any problems? Also, with this change to `numMaxLights` from 5 to 3, the register assignments for the light arrays are now off (in a way that won't cause functional problems, just wasted space).

I wonder if it's better to revert this change for now and deal with it in a follow-up issue? If not, I recommend to make the change from 5 to 3 more broadly.

Reply:
You are right, I will revert. I want to remove the restriction for 3 lights in the future anyway.